include/deal.II/lac/la_parallel_vector.templates.h

```diff
@@ -635,6 +720,47 @@ namespace parallel
 
     template <typename Number>
     void
+    Vector<Number>::import(const ReadWriteVector<Number>                  &V,
+                           VectorOperation::values                         operation,
+                           std_cxx11::shared_ptr<const CommunicationPatternBase> communication_pattern)
+    {
+      // If no communication pattern is given, create one. Otherwise, use the
+      // given one.
+      std_cxx11::shared_ptr<const Utilities::MPI::Partitioner> comm_pattern;
```

Comment:
In the epetra vector you store a partitioner. We should introduce that optimization here as well but for the moment my aim was to discuss the interface, keep all backward functionality (that we should restrict somehow before the release) and see if this is a sustainable path. (The optimizations also include the remainder discussed in #2535 regarding fast change between this vector and a readwrite vector.)